plugins/spotify/src/main/java/de/unistuttgart/iaas/amyassist/amy/plugin/spotify/logic/PlayerLogic.java

```diff
@@ -275,7 +299,40 @@ public int setVolume(int volume) {
 	 * @return the volume
 	 */
 	public int getVolume() {
+		if (this.srListening) {
+			return this.currentVolume;
+		}
 		return this.spotifyAPICalls.getVolume();
 	}
 
+	/**
+	 * called when volume has to be changed
+	 * 
+	 * @param volume
+	 *            volume to be changed to
+	 */
+	private void changeVolume(int volume) {
+		this.currentVolume = volume;
+		if (!this.srListening) {
+			this.spotifyAPICalls.setVolume(volume);
+		}
+	}
+
+	/**
+	 * activate/Deactivate that the sr is currently active -> volume changes will be queued until sr is no longer
+	 * listening by using the srListening variabel
+	 * 
+	 * @param isSRListening
+	 *            State to change to
+	 */
+	public void setSRListening(boolean isSRListening) {
+		if (isSRListening) {
+			this.spotifyAPICalls.setVolume(0);
+			this.srListening = isSRListening;
+		} else {
+			this.srListening = isSRListening;
+			changeVolume(Math.max(0, Math.min(100, this.currentVolume)));
```

Comment:
Isn't it more useful if we save the current SpotifyVolume in a variable when the SRListening is active and after the SRListening is over we set it back to its previous value. Because actually it would be set to 0 if the value of currentVolume doesn't change.

Reply:
Agree. This would also help, if the volume was changed through a input other than Amy.

Reply:
Actually what I thought @astyz meant is something like:

    changeVolume(int vol) {
      if(sr.listening) {
         this.volume = vol;
      } else 
         this.setSpotifyVol(vol);
      }
    }

    getVolume() {
      if(sr.listening) {
       return this.volume;
      } else 
       return this.getSpotifyVol();
      }
    }

That is a bit simpler solution, then what you proposed and does not have the problems you stated.